docs/sphinx/examples/matrix_multiplication.rst

```diff
@@ -0,0 +1,107 @@
+..
+    Copyright (C) 2021 Dimitra Karatza
+
+    SPDX-License-Identifier: BSL-1.0
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _examples_matrix_multiplication:
```

Comment:
Somewhere in the example we should mention that this example is for demonstration purposes only, and that users should prefer to use optimized libraries like BLAS or blaze (https://bitbucket.org/blaze-lib/blaze, it even has an HPX backend!).